File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    case x**2 + y**2 do
      r2 when r2 > 100 -> 0 # outside target
      r2 when r2 > 25 and r2 <= 100 -> 1 # outer circle
      r2 when r2 > 1 and r2 <= 25 -> 5 # middle circle
      r2 when r2 >= 0 and r2 <= 1 -> 10  # inner circle
    end
  end
end

File: c/darts/darts.c
#include "darts.h"

int score(coordinate_t dart) {
    float x = dart.x, y = dart.y;
    // pythagoras to find length of distance from origin to dart
    double distance = sqrt((x * x) + (y * y));
    
    if (distance >= 0 && distance <= 1) {
        return 10;
    } else if (distance > 1 && distance <= 5) {
        return 5;
    } else if (distance > 5 && distance <= MAX_POINT) {
        return 1;
    }
    return 0;
}
